qa/tasks/pykmip.py

```diff
@@ -0,0 +1,520 @@
+    log.info("pykmip_url=%s", pykmip_url)
+    #fetching user_id of user that gets secrets for radosgw
+    token_req = httplib.HTTPConnection(keystone_host, keystone_port, timeout=30)
+    token_req.request(
+        'POST',
+        '/v2.0/tokens',
+        headers={'Content-Type':'application/json'},
+        body=json.dumps(
+            {"auth":
+             {"passwordCredentials":
+              {"username": rgw_user["username"],
+               "password": rgw_user["password"]
+              },
+              "tenantName": rgw_user["tenantName"]
+             }
+            }
+        )
+    )
+    rgw_access_user_resp = token_req.getresponse()
+    if not (rgw_access_user_resp.status >= 200 and
+            rgw_access_user_resp.status < 300):
+        raise Exception("Cannot authenticate user "+rgw_user["username"]+" for secret creation")
+    #    baru_resp = json.loads(baru_req.data)
+    rgw_access_user_data = json.loads(rgw_access_user_resp.read())
+    rgw_user_id = rgw_access_user_data['access']['user']['id']
+
+    if 'secrets' in cconfig:
+        for secret in cconfig['secrets']:
+            if 'name' not in secret:
+                raise ConfigError('pykmip.secrets must have "name" field')
+            if 'base64' not in secret:
+                raise ConfigError('pykmip.secrets must have "base64" field')
+            if 'tenantName' not in secret:
+                raise ConfigError('pykmip.secrets must have "tenantName" field')
+            if 'username' not in secret:
+                raise ConfigError('pykmip.secrets must have "username" field')
+            if 'password' not in secret:
+                raise ConfigError('pykmip.secrets must have "password" field')
+
+            token_req = httplib.HTTPConnection(keystone_host, keystone_port, timeout=30)
+            token_req.request(
+                'POST',
+                '/v2.0/tokens',
+                headers={'Content-Type':'application/json'},
+                body=json.dumps(
+                    {
+                        "auth": {
+                            "passwordCredentials": {
+                                "username": secret["username"],
+                                "password": secret["password"]
+                            },
+                            "tenantName":secret["tenantName"]
+                        }
+                    }
+                )
+            )
+            token_resp = token_req.getresponse()
+            if not (token_resp.status >= 200 and
+                    token_resp.status < 300):
+                raise Exception("Cannot authenticate user "+secret["username"]+" for secret creation")
+
+            token_data = json.loads(token_resp.read())
+            token_id = token_data['access']['token']['id']
+
+            key1_json = json.dumps(
+                {
+                    "name": secret['name'],
+                    "expiration": "2020-12-31T19:14:44.180394",
```

Comment:
what is this date literal (in the past) doing for us?

Reply:
(does it represent an expired key?)

Reply:
This code was inherited from task keystone.  You'll notice the rest of this method  is doing keystone things involving json & http, not kmip things.  And it starts off with a try/yield/finally block which in teuthlogy land is a standard no-op.  A later commit "rgw/kms/kmip - pykmip.py needs to make keys too." replaces all of this with code that actually works--and is much shorter.

A terrible way to fix this would be to go back and remove the placeholder code from the old commit.  It's terrible because this placeholder code in fact has the tested behavior that it does nothing without breaking (and yet is clearly Just Wrong), and any replacement code would (should) be tested to verify the same behavior.  Another way to fix this would be to squash this commit in with the earlier code - that's not bad except it loses the commit message which I think might actually be helpful but I don't think belongs inline with the code.

Reply:
ok